roles/orphan_delete/tasks/main.yml

```diff
@@ -0,0 +1,34 @@
+# Copyright (c) 2023 VEXXHOST, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+
+- name: Delete deployments
+  kubernetes.core.k8s:
+    name: testing
+    api_version: apps/v1
+    kind: Deployment
+    state: absent
+    delete_options:
+      propagationPolicy: Orphan
+  loop: {{ orphan_delete_deployments}}
+
+- name: Delete statefulsets
+  kubernetes.core.k8s:
+    name: testing
+    api_version: apps/v1
+    kind: StatefulSet
+    state: absent
+    delete_options:
+      propagationPolicy: Orphan
+  loop: {{ orphan_delete_statefulsets}}
```

Comment:
This should all be auto-detected.

Reply:
@mnaser i made some progress for auto detection
let me make a patch in the gerrit